### ISSUE TITLE
Properly treat the generation of duplicate path helpers

### ISSUE THREAD

**Issue by thiagomajesk** [tiagoengel/phoenix-jsroutes#15]
Hello @tiagoengel!

First off, thanks for the awesome work with this lib! 
It's so useful that I don't understand why this is not already part of Phoenix in some way.

I have a suggestion that consists of treating cases when the library finds a duplicate route, instead of silently failing. Take this route config as an example:

```elixir
get "/users", UserController, :index
get "/users-fallback", UserController, :index
```

Even though this is kinda valid in Phoenix, it throws a cryptic error message that helps no one understand what's happening:

```javascript
phoenix-jsroutes.js:19 Uncaught TypeError: Cannot read property 'PhoenixJsRoutes' of undefined
    at phoenix-jsroutes.js:19
    at Module.<anonymous> (phoenix-jsroutes.js:8)
    at Module../static/js/phoenix-jsroutes.js (app.js:61222)
    at __webpack_require__ (bootstrap:19)
    at Module../js/components/item-selection.js (app.js:266)
    at __webpack_require__ (bootstrap:19)
    at Module../js/app.js (app.js:1)
    at __webpack_require__ (bootstrap:19)
    at Object.0 (app.js:61233)
    at __webpack_require__ (bootstrap:19)
```

The aforementioned route config will generate the same path helper for both routes, but Phoenix only considers the last one. 
So, `user_path(conn, :index)` actually points to the `~/users-fallback` path. If I need both routes I can easily rename the path helpers using the `:as` option, but the lib doesn't do anything to warn us about it. Depending on the number of routes the application has it can take several minutes to find the problematic one.

Going forward, my suggestion to improve usability and making users fall into the "Pit of Success", is to either: 

1) Raise an exception whenever we find a duplicate route;
2) Mimic Phoenix's behavior (only registering the last route);
3) Allow the user to set a global option for the duplicate strategy (raise or override)

**Comment by tiagoengel** [comment]
Hey @thiagomajesk, first of all, you have a very cool name 😛 . Thanks for raising this.

The current error looks like a bug to me, I imagine it's because we are generating two functions with the same name and the JS module is not compiling.

My first inclination would be to implement your second suggestion, and follow Phoenix's behaviour. I think It's the less surprising outcome to users.

And in the end if both paths are calling the same resource it doesn't really matter which one of the two we are using.

**Comment by tiagoengel** [comment]
Hey @thiagomajesk,  in which version of the lib + phoenix are you seeing this error? I just tried this locally in the example app and I can't reproduce this error.

**Comment by thiagomajesk** [comment]
Hi @tiagoengel! Sorry for the delay...

I've tested with both Phoenix 1.4.* and 1.5.* using version 1.0.0

**Comment by tiagoengel** [comment]
Do you mind creating a git repo to reproduce this? I'm not able to reproduce it just adding duplicated routes.

**Comment by thiagomajesk** [comment]
@tiagoengel Hey man! I'm sorry for the delay, I had a lot on my plate this week.

Weirdly enough, I just created a project from scratch and the error message did not happen whatsoever - even though there was, in fact, two identical object keys (which is still a valid js syntax). This is the route config I've used:

**router.ex**
```elixir
resources "/foo", FooController, except: [:update]
# forcing path helpers with the same name
put "/foo", FooController, :update
patch "/foo", FooController, :update
```

**app.js**
```javascript
import routes from '../static/js/phoenix-jsroutes'
// preventing webpack tree-shaking by actually using the function
console.log("foo update route", routes.fooUpdate(1))
```
I'm gonna ask a coworker who first spotted this if we can reduce our current project to have a more reliable repro.

**PS.:** I wouldn't be surprised if we discover that this is something related to Webpack itself and not your lib :see_no_evil: 

Cheers!

**Comment by rafaeliga** [comment]
hey @tiagoengel, this is the git repo with the error: https://github.com/rafaeliga/phoenix-jsroutes-error

Commits have the "steps" to reproduce the error:

1. create a new phoenix project
2. install phoenix-jsroutes
3. customize .babelrc with corejs 3

We couldn't find the source of the issue... maybe function name on corejs? some strict config?

Thanks

**Comment by tiagoengel** [comment]
Thanks @rafaeliga. I'll have a look as soon as I can.

**Comment by tiagoengel** [comment]
Hey @rafaeliga and @thiagomajesk. Sorry for the delay, I had a look at this today and found two issues. 
1. There is a compilation error because `corejs` is missing, installing `corejs3` fixes it.
2. The modules are not being defined. Fore some reason when corejs 3 is used the way webpack define the modules is different and it doesn't work with the current implementation.

I'm not sure how to fix this to be honest, I've found a few hacks that work but it seems to be quite fragile. I'm thinking of introducing a new breaking change to drop support to all other ways of importing this (global, AMD) and using a normal export syntax, that webpack would then handle correctly.

Let me know if you a have any other ideas.

**Comment by thiagomajesk** [comment]
Hey @tiagoengel! Thanks for the feedback.

In our original project, we have corejs ^3.6.4 installed, @rafaeliga can you take a look and update our repro?

Honestly, I haven't used RequireJs for the past 5 years, so I wouldn't miss it. 
Besides that, I think that using Babel with Webpack is going to be more common because of Phoenix's default setup.

I like the idea of releasing a major version with only `import/ export`, anyone else who wants to use AMD can use the legacy version. Also, I think this is more future-proof and less error-prone with modern tools setups.

**Comment by tiagoengel** [comment]
Just released version 2.0.0 with this change @thiagomajesk @rafaeliga
